### PR TITLE
XWIKI-15444: Export UI does not export child pages if there are many (and tree has pagination)

### PR DESCRIPTION
**Pull request by surli**
* Fix XARExportTest flickering test

### CODE REVIEW

**Comment by surli** [comment]
@mflorea this one's ready for review and should fix the XARExportTest which is failing on the CI.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java [hunk 1]
```diff
@@ -49,8 +51,15 @@ public ExportModal()
      */
     public OtherFormatPane openOtherFormatPane()
     {
-        getDriver().findElementByLinkText(OTHER_FORMAT_PANE_LINK).click();
-        getDriver().waitUntilElementIsVisible(By.cssSelector("#exportModelOtherCollapse .panel-body"));
+        WebElement linkPane = getDriver().findElementByLinkText(OTHER_FORMAT_PANE_LINK);
+        linkPane.click();
+
+        // sometimes it looks like the first click does not work.
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java]: This looks like a hack. Would be nice to understand why the first click doesn't work. Maybe the link is moving (e.g. if there is some animation). AFAIK the click is done in 2 steps:

* it first gets the rectangular area that fits the button / link and computes the center point
* sends a click event in the center point

This kind of simulates the user behavior:

* you locate the button visually on the page and move the mouse over it
* you click

The issue is that the button / link may be moved between the steps, so you end up clicking on something else.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java [hunk 2]
```diff
@@ -49,8 +51,15 @@ public ExportModal()
...
+        if (linkPane.getAttribute("class").contains("collapsed")) {
+            linkPane.click();
+        }
+        By paneContent = By.className(CONTAINER_TREE_CLASS);
+        getDriver().waitUntilElementIsVisible(paneContent);
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java]: AFAICS the pane content is displayed right away but there is some animation and some opacity transition. If you check the class it changes from:

    collapse -> collapsing -> collapse in

So you should probably wait until the pane content has class "collapse in".

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java (hunk 1 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java]: It's absolutely a hack: I really don't understand why sometimes the click don't work the first time. Might be related to the fading of the modal (even though it should be managed in the modal PO). Might be the accordion, but don't see why.
It's the only solution I found to make the test passing 10 times on 10.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java (hunk 2 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-pageobjects/src/main/java/org/xwiki/flamingo/sking/test/po/ExportModal.java]: I can change that, but anyway the bug is not related to that: I got a 20 sec timeout and when looking on the screenshot of the failing test, you can see that the pane is not displayed at all, as if the click hasn't been done.

**Pull request merged**